lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -128,23 +139,17 @@ namespace Js
         }
         else
         {
-            int foundIndex = -1;
-            bool parentFound = this->otherParents != nullptr && this->otherParents->MapUntil([&](int index, RecyclerWeakReference<ArrayBufferParent>* item)
+            int foundCount = 0;
+            this->otherParents->MapAndRemoveIf([&](RecyclerWeakReference<ArrayBufferParent>* item)->bool
             {
-                if (item->Get() == parent)
+                if(item->Get() == nullptr)
                 {
-                    foundIndex = index;
+                    foundCount++;
                     return true;
                 }
                 return false;
-
             });
-
-            if (parentFound)
-            {
-                this->otherParents->RemoveAt(foundIndex);
-            }
-            else
+            if (foundCount != 1)
```

Comment:
>foundCount [](start = 16, length = 10)

Confused with this condition. Above IIRC `MapAndRemoveIf` will continue iterating as long as `item-Get() == nullptr` in which case `foundCount can be > 1` ?